inbm-lib/inbm_common_lib/utility.py

```diff
@@ -180,11 +180,11 @@ def canonicalize_uri(url: str) -> CanonicalUri:
```

Comment:
Ensure that the `canonicalize_uri` function is robust against various edge cases and unexpected inputs, possibly by adding more unit tests. [important]